paddle/majel/place.h

```diff
@@ -14,33 +14,24 @@ struct CpuPlace {
 };
 
 struct GpuPlace {
-  GpuPlace(int d) : device(d) {}
+  GpuPlace() {}
 
   // needed for variant equality comparison
-  inline bool operator==(const GpuPlace& o) const { return device == o.device; }
+  inline bool operator==(const GpuPlace&) const { return true; }
 
-  inline bool operator!=(const GpuPlace& o) const { return !(*this == o); }
-
-  GpuPlace() : GpuPlace(0) {}
-  int device;
```

Comment:
I think communications between devices are costful, one neural network will be run on one device. If we are going to support OpenCL/FPGA, we can just define ```Place``` as follows:
```
typedef boost::variant<CpuPlace, CudaPlace, OpenclPlace, FpgaPlace> Place;
```
Then, we implement methods, such as malloc/free of corresponding device.